### PR TITLE
WT-10971 Fix truncate for tiered storage.

### PR DESCRIPTION
**Pull request by ddanderson**
* Fixed a bug in tiered truncate that tried to truncate the file:....-0000001.wtobj uri
* Enable the vast group of truncate tests that were previous disabled in the tiered hook.
* Individually skip or short-circuit a few truncate tests that rely on fast truncates.
* Skip tests with named checkpoints for tiered storage.
* Improve wttest so that stdout is line buffered, and we have verbose output when a test begins.
* If a test calls dropUntilSuccess, it's an indication that the test really needs the namespace cleared out, don't do it for tiered storage, as most storage source module refuse to implement cloud removal.

### CODE REVIEW

**Comment by ddanderson** [comment]
@pmacko86 @keitharnoldsmith, can you please review?  This is the "improved" version of PR #9133 which you both reviewed.  That one needed to be reverted, as it failed the `unit-test-hook-tiered-s3`, which isn't part of PR testing.  (I added it to the testing for this one).  See the ticket or the PR description for a description of this change.

The basic problem with the other PR was that it enabled some tests that call the function `WiredTigerTestCase.dropUntilSuccess`, and that's typically an indication that the test wants all related artifacts removed, (because it wants to do a variation of the same test with the same table names).  I had turned on drop's `"remove_shared=true"` flag when that function is called.  But that flag is only supported by `dir_store`, not `s3_store` or the others.  So the tests fail when run under s3.

I think not supporting removal of cloud files is a good protective stance at this time, so I didn't want to change that in s3 et al.  So instead, I chose to skip tests that call `dropUntilSuccess` when running under the tiered hook.

For future, we talked about "packaging multiple objects" in the cloud (PM-3389).  Depending on how that is implemented, we may end up with unique cloud names, in which case we can remove the skip and those tests should probably work again.

Timing: I'm "mostly" out until July 10, so certainly no rush on the review!  Thanks.

File: test/suite/hook_tiered.py [hunk 1]
```diff
@@ -219,11 +238,9 @@ def session_create_replace(orig_session_create, session_self, uri, config):
 # do statistics on (tiered) table data sources, as that is not yet supported.
 def session_open_cursor_replace(orig_session_open_cursor, session_self, uri, dupcursor, config):
     if uri != None and (uri.startswith("statistics:table:") or uri.startswith("statistics:file:")):
-        testcase = WiredTigerTestCase.currentTestCase()
-        testcase.skipTest("statistics on tiered tables not yet implemented")
+        skipTest("statistics on tiered tables not yet implemented")
```
Comment (pmacko86) [inline test/suite/hook_tiered.py:241]: I like the simplification coming from the refactoring here.

**Review by pmacko86 — APPROVED** [review]
LGTM. I don't fully understand how hooks work yet, but overall looks good!

**Comment by ddanderson** [comment]
evergreen merge

**Pull request merged**